src/support.js

```diff
@@ -575,6 +575,13 @@ function loadWebAssemblyModule(binary, flags) {
         assert(table.get(tableBase + i) !== undefined, 'table entry was not filled in');
       }
 #endif
+      if (!functionsInTableMap) var functionsInTableMap = new WeakMap();
+      for (var i = tableBase + 1; i < table.length; i++) {
+        var item = table.get(i);
+        if (item) {
+          functionsInTableMap.set(item, i);
+        }
+      }
```

Comment:
How about just doing `functionsInTableMap = undefined;` to invalidate the entire map each time a side module is loaded.

Then the next time addFunctionWasm is called it will be recreated from scratch?

Reply:
That's an interesting idea, and it is simpler, but I think the inefficiency would be annoying.

See my other comment, I think we can reduce to a single function call from here, after some refactoring in the function map code. I think that will keep it pretty simple too.